roundtripper/roundtripper.go

```diff
@@ -39,27 +40,28 @@ const (
 	GithubAppInstallationID = "GITHUB_APP_INSTALLATION_ID"
 )
 
+func readGitHubToken() (string, bool) {
+	for _, name := range GithubAuthTokens {
+		if token, exists := os.LookupEnv(name); exists && token != "" {
+			return token, exists
+		}
+	}
+	return "", false
+}
+
 // NewTransport returns a configured http.Transport for use with GitHub.
 func NewTransport(ctx context.Context, logger *zap.SugaredLogger) http.RoundTripper {
 	transport := http.DefaultTransport
 
-	if token := os.Getenv(GithubAuthToken); token != "" {
-		// Use GitHub PAT
-		transport = githubrepo.MakeGitHubTransport(transport, strings.Split(token, ","))
-	} else if keyPath := os.Getenv(GithubAppKeyPath); keyPath != "" { // Also try a GITHUB_APP
```

Comment:
if I understand comment https://github.com/ossf/scorecard/commit/7cf56e04afc54d1f894a130270dfecf951ae23e0, we ca no longer support app token because of graphQl.

Reply:
What I meant was that you cannot use graphQL without authorization headers of some sort. It could be PATs or GH app. So we should continue supporting GH apps.

Reply:
gocha.